Mathlib/Topology/MetricSpace/DilationEquiv.lean

```diff
@@ -176,6 +178,46 @@ def toPerm : (X ≃ᵈ X) →* Equiv.Perm X where
 theorem coe_pow (e : X ≃ᵈ X) (n : ℕ) : ⇑(e ^ n) = e^[n] := by
   rw [← coe_toEquiv, ← toPerm_apply, map_pow, Equiv.Perm.coe_pow]; rfl
 
+-- TODO: Once `IsometryEquiv` follows the `*EquivClass` pattern, replace this with an instance
+-- of `DilationEquivClass` assuming `IsometryEquivClass`.
+/-- Every isometry equivalence is a dilation equivalence of ratio `1`. -/
+def _root_.IsometryEquiv.toDilationEquiv (e : X ≃ᵢ Y) : X ≃ᵈ Y where
+  edist_eq' := ⟨1, one_ne_zero, by simpa using e.isometry⟩
+  __ := e.toEquiv
+
+@[simp]
+lemma _root_.IsometryEquiv.toDilationEquiv_apply (e : X ≃ᵢ Y) (x : X) :
+    e.toDilationEquiv x = e x :=
+  rfl
+
+@[simp]
+lemma _root_.IsometryEquiv.toDilationEquiv_symm (e : X ≃ᵢ Y) :
+    e.toDilationEquiv.symm = e.symm.toDilationEquiv :=
+  rfl
+
+@[simp]
+lemma _root_.IsometryEquiv.toDilationEquiv_toDilation (e : X ≃ᵢ Y) :
+    (e.toDilationEquiv.toDilation : X →ᵈ Y) = e.isometry.toDilation :=
+  rfl
+
+@[simp]
+lemma _root_.IsometryEquiv.toDilationEquiv_ratio (e : X ≃ᵢ Y) : ratio e.toDilationEquiv = 1 := by
+  rw [← ratio_toDilation, IsometryEquiv.toDilationEquiv_toDilation, Isometry.toDilation_ratio]
+
+/-- Reinterpret a `DilationEquiv` as a homeomorphism. -/
+def toHomeomorph (e : X ≃ᵈ Y) : X ≃ₜ Y where
+  continuous_toFun := Dilation.toContinuous e
+  continuous_invFun := Dilation.toContinuous e.symm
+  __ := e.toEquiv
```

Comment:
Do we really have no `HomeomorphClass` ????

Reply:
We don't, no. I was a bit annoyed by this too, but not so much that I wanted to go write it immediately.